azure/services/managedclusters/spec.go

```diff
@@ -581,3 +591,20 @@ func computeDiffOfNormalizedClusters(managedCluster containerservice.ManagedClus
 	diff := cmp.Diff(clusterNormalized, existingMCClusterNormalized)
 	return diff
 }
+
+func getIdentityProfile(identityProfile *infrav1.IdentityProfile) (managedClusterIdentity *containerservice.ManagedClusterIdentity, err error) {
+	if identityProfile.Type != "" {
+		managedClusterIdentity = &containerservice.ManagedClusterIdentity{
```

Comment:
nit: removing most of this level of indentation might make this a bit more readable:
```suggestion
	if identityProfile.Type == "" {
		return
	}
	managedClusterIdentity = &containerservice.ManagedClusterIdentity{
```

Reply:
^ in addition to the above change, we might want to add an error as well.

Reply:
we are returning an error it's just that this func is using named response variables so things are maybe a little cute (but using a named `*containerservice.ManagedClusterIdentity` response variable in particular does clean up the code a bit)